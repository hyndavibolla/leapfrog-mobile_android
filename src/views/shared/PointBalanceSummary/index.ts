export { default as PointBalanceSummary } from './PointBalanceSummary';
export type { Props as PointBalanceSummaryProps } from './PointBalanceSummary';
