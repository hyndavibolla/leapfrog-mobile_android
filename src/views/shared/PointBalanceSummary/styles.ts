import { css } from '@emotion/native';

import { COLOR, FONT_SIZE } from '../../../constants/styles';

export const styles = {
  containerGeneral: css`
    flex-direction: column;
    justify-content: space-between;
    background-color: ${COLOR.WHITE};
    padding: 0;
  `,
  backgroundContainer: css`
    background-color: transparent;
    height: 72px;
    width: 100%;
  `,
  gradient: css`
    bottom: 0;
    width: 100%;
    height: 72px;
    border-top-left-radius: 8px;
    border-top-right-radius: 8px;
    padding: 4px;
  `,
  containerPoints: css`
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    height: 100%;
    padding: 12px 16px 20px 20px;
    border-top-left-radius: 8px;
    border-top-right-radius: 8px;
  `,
  containerAvailable: css`
    flex-direction: column;
    justify-content: space-between;
  `,
  containerPending: css`
    flex-direction: column;
    justify-content: space-between;
  `,
  containerPendingTitle: css`
    flex-direction: row;
    align-items: center;
  `,
  eachPointsContainer: css`
    flex-direction: row;
    align-items: center;
  `,
  icon: css`
    align-items: center;
    justify-content: center;
    border-radius: 50px;
    margin-right: 4px;
  `,
  textTitle: css`
    color: ${COLOR.WHITE};
    font-size: ${FONT_SIZE.TINY};
    text-transform: uppercase;
  `,
  textPoints: css`
    color: ${COLOR.WHITE};
    font-size: ${FONT_SIZE.BIG};
    line-height: 28px;
  `,
  containerMember: css`
    background-color: ${COLOR.WHITE};
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    padding: 20px;
    border-bottom-left-radius: 8px;
    border-bottom-right-radius: 8px;
  `,
  containerBarcode: css`
    flex-direction: row;
    align-items: center;
  `,
  textMember: css`
    color: ${COLOR.DARK_GRAY};
    font-size: ${FONT_SIZE.PETITE};
    text-transform: uppercase;
    margin-left: 8px;
  `,
  pointsContainer: css`
    flex-direction: row;
    align-items: center;
  `,
  buttonIcon: css`
    margin-right: 4px;
  `
};
