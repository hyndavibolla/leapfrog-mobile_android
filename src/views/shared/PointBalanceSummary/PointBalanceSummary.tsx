import React, { memo } from 'react';
import { TouchableHighlight, View, ImageBackground } from 'react-native';
import { Text } from '../Text';

import { styles } from './styles';
import { Icon } from '_commons/components/atoms/Icon';
import ClaimIcon from '_assets/button/claimIcon.svg';
import { Card } from '../Card';
import { formatNumber } from '_utils/formatNumber';
import { useTestingHelper } from '_utils/useTestingHelper';
import { COLOR, colorWithOpacity, FONT_FAMILY, FONT_SIZE, ICON } from '_constants';
import LinearGradient from 'react-native-linear-gradient';

export interface Props {
  onPressTop?: () => void;
  onPressBottom?: () => void;
  availablePoints?: number;
  pendingPoints?: number;
}

export const PointBalanceSummary = ({ onPressTop, onPressBottom, availablePoints = 0, pendingPoints = 0 }: Props) => {
  const { getTestIdProps } = useTestingHelper('point-balance-summary');
  return (
    <Card style={styles.containerGeneral}>
      <LinearGradient
        start={{ x: 0, y: 0.75 }}
        end={{ x: 0, y: 1 }}
        style={styles.gradient}
        colors={[colorWithOpacity(COLOR.PRIMARY_BLUE, 80), COLOR.PRIMARY_BLUE]}
        {...getTestIdProps('gradient')}
      >
        <ImageBackground
          style={styles.backgroundContainer as any}
          imageStyle={{ flex: 1, resizeMode: 'repeat' }}
          source={require('_assets/shared/pointBalancePatternAsset.png')}
        >
          <TouchableHighlight {...getTestIdProps('top-btn')} onPress={onPressTop} underlayColor="transparent" activeOpacity={1}>
            <View style={styles.containerPoints}>
              <View style={styles.containerAvailable}>
                <Text font={FONT_FAMILY.BOLD} style={styles.textTitle}>
                  Available Points
                </Text>
                <View style={styles.pointsContainer}>
                  <ClaimIcon style={styles.buttonIcon} width={20} height={20} />
                  <Text font={FONT_FAMILY.HEAVY} style={styles.textPoints} {...getTestIdProps('available-points-label')}>
                    {formatNumber(availablePoints)}
                  </Text>
                </View>
              </View>
              <View style={styles.containerPending}>
                <View style={styles.containerPendingTitle}>
                  <Text font={FONT_FAMILY.BOLD} style={styles.textTitle}>
                    Pending
                  </Text>
                </View>
                <View style={styles.pointsContainer}>
                  <ClaimIcon style={styles.buttonIcon} width={20} height={20} />
                  <Text font={FONT_FAMILY.HEAVY} style={styles.textPoints} {...getTestIdProps('pending-points-label')}>
                    {formatNumber(pendingPoints)}
                  </Text>
                </View>
              </View>
              <Icon name={ICON.BRACKET_RIGHT} size={FONT_SIZE.REGULAR} color={COLOR.WHITE} />
            </View>
          </TouchableHighlight>
        </ImageBackground>
      </LinearGradient>
      <TouchableHighlight {...getTestIdProps('bottom-btn')} onPress={onPressBottom} underlayColor="transparent" activeOpacity={1}>
        <View style={styles.containerMember}>
          <View style={styles.containerBarcode}>
            <Icon name={ICON.BAR_CODE} size={FONT_SIZE.HUGE} color={COLOR.DARK_GRAY} />
            <Text font={FONT_FAMILY.BOLD} style={styles.textMember}>
              Your member number and pin
            </Text>
          </View>

          <Icon name={ICON.BRACKET_RIGHT} size={FONT_SIZE.SMALL} color={COLOR.DARK_GRAY} />
        </View>
      </TouchableHighlight>
    </Card>
  );
};

export default memo(PointBalanceSummary);
