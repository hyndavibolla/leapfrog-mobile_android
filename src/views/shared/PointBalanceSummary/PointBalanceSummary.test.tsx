import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import PointBalanceSummary, { Props } from './PointBalanceSummary';

describe('PointBalanceSummary', () => {
  let onPressTop: () => void;
  let onPressBottom: () => void;
  let props: Props;
  beforeEach(() => {
    onPressTop = jest.fn();
    onPressBottom = jest.fn();
    props = {
      onPressTop,
      onPressBottom,
      availablePoints: 37500,
      pendingPoints: 15320
    };
  });

  it('should render with defaults', () => {
    const { toJSON } = render(<PointBalanceSummary onPressTop={props.onPressTop} onPressBottom={props.onPressBottom} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should be pressable on the top button', () => {
    const { getByTestId } = render(
      <PointBalanceSummary
        onPressTop={props.onPressTop}
        onPressBottom={props.onPressBottom}
        availablePoints={props.availablePoints}
        pendingPoints={props.pendingPoints}
      />
    );
    fireEvent.press(getByTestId('point-balance-summary-top-btn'));
    expect(props.onPressTop).toBeCalled();
  });

  it('should be pressable on the bottom button', () => {
    const { getByTestId } = render(
      <PointBalanceSummary
        onPressTop={props.onPressTop}
        onPressBottom={props.onPressBottom}
        availablePoints={props.availablePoints}
        pendingPoints={props.pendingPoints}
      />
    );
    fireEvent.press(getByTestId('point-balance-summary-bottom-btn'));
    expect(props.onPressBottom).toBeCalled();
  });
});
