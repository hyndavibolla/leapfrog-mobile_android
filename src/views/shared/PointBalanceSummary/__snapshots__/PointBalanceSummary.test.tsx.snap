// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PointBalanceSummary should render with defaults 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "elevation": 3,
        "shadowColor": "#262C2F",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.1,
      },
      Object {
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      },
      Object {
        "backgroundColor": "#FFFFFF",
        "flexDirection": "column",
        "justifyContent": "space-between",
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      },
    ]
  }
>
  <BVLinearGradient
    accessibilityLabel="point-balance-summary-gradient"
    accessible={true}
    colors={
      Array [
        2147509964,
        4278216396,
      ]
    }
    endPoint={
      Object {
        "x": 0,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0.75,
      }
    }
    style={
      Object {
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "bottom": 0,
        "height": 72,
        "paddingBottom": 4,
        "paddingLeft": 4,
        "paddingRight": 4,
        "paddingTop": 4,
        "width": "100%",
      }
    }
    testID="point-balance-summary-gradient"
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "height": 72,
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/shared/pointBalancePatternAsset.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 72,
              "width": "100%",
            },
            Object {
              "flex": 1,
              "resizeMode": "repeat",
            },
          ]
        }
      />
      <View
        accessibilityLabel="point-balance-summary-top-btn"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="point-balance-summary-top-btn"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "flexDirection": "row",
              "height": "100%",
              "justifyContent": "space-between",
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 16,
              "paddingTop": 12,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              font="SFProDisplay-Bold"
              style={
                Array [
                  Object {
                    "color": "#909BA0",
                    "fontFamily": "SFProDisplay-Bold",
                    "fontSize": 16,
                    "lineHeight": 22,
                  },
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 12,
                    "textTransform": "uppercase",
                  },
                ]
              }
            >
              Available Points
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              ClaimIcon
              <Text
                accessibilityLabel="point-balance-summary-available-points-label"
                accessible={true}
                font="SFProDisplay-Heavy"
                style={
                  Array [
                    Object {
                      "color": "#909BA0",
                      "fontFamily": "SFProDisplay-Heavy",
                      "fontSize": 16,
                      "lineHeight": 22,
                    },
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 24,
                      "lineHeight": 28,
                    },
                  ]
                }
                testID="point-balance-summary-available-points-label"
              >
                0
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                font="SFProDisplay-Bold"
                style={
                  Array [
                    Object {
                      "color": "#909BA0",
                      "fontFamily": "SFProDisplay-Bold",
                      "fontSize": 16,
                      "lineHeight": 22,
                    },
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 12,
                      "textTransform": "uppercase",
                    },
                  ]
                }
              >
                Pending
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              ClaimIcon
              <Text
                accessibilityLabel="point-balance-summary-pending-points-label"
                accessible={true}
                font="SFProDisplay-Heavy"
                style={
                  Array [
                    Object {
                      "color": "#909BA0",
                      "fontFamily": "SFProDisplay-Heavy",
                      "fontSize": 16,
                      "lineHeight": 22,
                    },
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 24,
                      "lineHeight": 28,
                    },
                  ]
                }
                testID="point-balance-summary-pending-points-label"
              >
                0
              </Text>
            </View>
          </View>
          <Text
            accessibilityLabel="icon-text"
            accessible={true}
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "sywmax",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="icon-text"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </BVLinearGradient>
  <View
    accessibilityLabel="point-balance-summary-bottom-btn"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="point-balance-summary-bottom-btn"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          accessibilityLabel="icon-text"
          accessible={true}
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontSize": 26,
              },
              undefined,
              Object {
                "fontFamily": "sywmax",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="icon-text"
        >
          
        </Text>
        <Text
          font="SFProDisplay-Bold"
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Bold",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Object {
                "color": "#909BA0",
                "fontSize": 14,
                "marginLeft": 8,
                "textTransform": "uppercase",
              },
            ]
          }
        >
          Your member number and pin
        </Text>
      </View>
      <Text
        accessibilityLabel="icon-text"
        accessible={true}
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "sywmax",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="icon-text"
      >
        
      </Text>
    </View>
  </View>
</View>
`;
